components/rule-mgt/org.wso2.carbon.identity.rule.metadata/src/main/java/org/wso2/carbon/identity/rule/metadata/config/FlowConfig.java

```diff
@@ -90,4 +104,81 @@ private static void validateFlow(String flowName) throws RuleMetadataConfigExcep
             throw new RuleMetadataConfigException("Invalid flow: " + flowName);
         }
     }
+
+    /**
+     * Resolve the field definition from the field definition config.
+     *
+     * @param fieldDefinitionConfig FieldDefinitionConfig
+     * @param entry                 Entry
+     * @return FieldDefinition
+     * @throws RuleMetadataConfigException If an error occurs while resolving the field definition
+     */
+    private static FieldDefinition resolveFieldDefinition(FieldDefinitionConfig fieldDefinitionConfig,
+                                                          Map<String, Object> entry)
+            throws RuleMetadataConfigException {
+
+        if (!entry.containsKey("fieldName")) {
+            throw new RuleMetadataConfigException("'fieldName' is required for a field");
+        }
+
+        String fieldName = entry.get("fieldName").toString();
+        if (!fieldDefinitionConfig.getFieldDefinitionMap().containsKey(fieldName)) {
+            throw new RuleMetadataConfigException("Invalid field: " + fieldName);
+        }
+
+        return fieldDefinitionConfig.getFieldDefinitionMap().get(fieldName);
+    }
+
+    /**
+     * Get the updated field if the field is overridden.
+     *
+     * @param mapper           ObjectMapper
+     * @param overrides        Overrides
+     * @param fieldDefinition  FieldDefinition
+     * @return Updated field
+     */
+    private static Field getUpdatedField(ObjectMapper mapper, Map<?, ?> overrides, FieldDefinition fieldDefinition) {
+
+        if (!overrides.containsKey("field")) {
+            return fieldDefinition.getField();
+        }
```

Comment:
should this be field or displayName

Reply:
Here the expectation is the user to provide whole updated field details, in the overrides object.

example :

```
"preUpdatePassword": [
    {
      "fieldName": "flow",
      "overrides": {
        "field": {
          "name": "flow",
          "displayName": "flow"
        },
        "value": {
          "values": [
            {
              "name": "admin_initiated_password_reset",
              "displayName": "Admin initiated password reset"
            }
          ]
        }
      }
    }
  ]
```

Reply:
And the user can refrain from providing that. At that scenario, we will override the values only.

Reply:
I think that scenario will not be needed.
So let's allow selectively. For the time being as it is values, let them oveeride that. and the other case it update just display name. Let's provide the functionality for that as well

Reply:
Lets only allow to override the displayName only in the `field` object